### PR TITLE
Features/form

### PR DESCRIPTION
**Pull request by Canfus**
Add components using with form.

### CODE REVIEW

File: frontend/src/shared/ui/select/select.tsx [hunk 1]
```diff
@@ -43,23 +47,56 @@ export const Select = forwardRef<React.ElementRef<typeof Trigger>, SelectProps>(
...
         return displayValue(selectedValue);
       }
 
       return selectedValue.label;
-    }, [displayValue, placeholder, selectedValue]);
+    }, [displayValue, selectedValue]);
+
+    const onChange = useCallback(
+      (newValue: string) => {
+        if (isStringEmpty(newValue)) {
```
Comment (Canfus) [inline frontend/src/shared/ui/select/select.tsx:64]: @worldspawn-web we need to add a lot a specific test cases, cuz this library was override our value by empty string on each choose.

File: frontend/src/shared/ui/select/select.tsx [hunk 2]
```diff
@@ -43,23 +47,56 @@ export const Select = forwardRef<React.ElementRef<typeof Trigger>, SelectProps>(
...
+          return;
+        }
```
Comment (Canfus) [inline frontend/src/shared/ui/select/select.tsx:66]: @worldspawn-web we need to add a lot a specific test cases, cuz this library was override our value by empty string on each choose.
image

Comment (worldspawn-web) [reply frontend/src/shared/ui/select/select.tsx:66]: If I got you right, then it works fine.

File: frontend/src/shared/widget/text-field/text-field.tsx
Comment (worldspawn-web) [inline frontend/src/shared/widget/text-field/text-field.tsx:1]: We have two problems here with `Text Field`, but it's more about validation.

1) **Form says nothing if everything is right, except email**. If you remove additional whitespace, error messages will appear as they must.
Screenshot 2024-09-02 212336

2) **Wrong message error**. The problem is that if the password and password re-checker are not equal, you will receive a wrong error message. It says that there is no upper letter, but in reality they just don't match each other. We must give no hints on that input.

Screenshot 2024-09-02 212224

File: frontend/src/stories/form/text-field/text-field.schema.ts
Comment (worldspawn-web) [inline frontend/src/stories/form/text-field/text-field.schema.ts:1]: I suppose that you will have to edit schema, Read the issues above.

**Review by worldspawn-web — CHANGES_REQUESTED** [review]
We have a couple of validation bugs, but other than that it's good.

File: frontend/src/stories/form/text-field/text-field.schema.ts
Comment (Canfus) [reply frontend/src/stories/form/text-field/text-field.schema.ts:1]: Validation issue 💀
It works good, I wrote some object calls `validators`, it have a lot of field validation, like stringRequiredValidationSchema and so on.
It should be work fine

File: frontend/src/shared/widget/text-field/text-field.tsx
Comment (Canfus) [reply frontend/src/shared/widget/text-field/text-field.tsx:1]: Of case 2 we need to check equal fields inside a schema.
In example:
```ts
object({})
  .refine((schema) => schema.password === schema.repeatPassword, "Password doesn't match");
```

**Comment by worldspawn-web** [comment]
@Canfus will you finish this one? ☠️

**Comment by Canfus** [comment]
> @Canfus will you finish this one? ☠️ 

Its done, it will be fixed by zod validation by side

**Review by worldspawn-web — APPROVED** [review]
Let's suppose that it actually works and I actually tested that in 2025...
